File: tck/scala-tck/src/main/scala/kalix/tck/model/valueentity/ValueEntityTwoEntity.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.tck.model.valueentity

import kalix.scalasdk.valueentity.ValueEntity

/** A value entity. */
class ValueEntityTwoEntity extends AbstractValueEntityTwoEntity {
  override def emptyState: Persisted = Persisted.defaultInstance

  override def call(currentState: Persisted, request: Request): ValueEntity.Effect[Response] =
    effects.reply(Response.defaultInstance)
}

File: tck/java-tck/src/main/java/kalix/tck/model/valueentity/ValueEntityTwoEntity.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.tck.model.valueentity;

import kalix.javasdk.valueentity.ValueEntityContext;
import kalix.tck.model.valueentity.ValueEntityApi.Request;
import kalix.tck.model.valueentity.ValueEntityApi.Response;

public class ValueEntityTwoEntity extends AbstractValueEntityTwoEntity {

  public ValueEntityTwoEntity(ValueEntityContext context) {}

  public Effect<Response> call(Persisted state, Request request) {
    return effects().reply(Response.getDefaultInstance());
  }

  @Override
  public Persisted emptyState() {
    return null;
  }
}
